addons/hr_holidays/tests/test_allocations.py

```diff
@@ -177,3 +179,41 @@ def test_disallowed_change_allocation_with_overlapping_allocations(self):
             allocation_one.write({'number_of_days_display': 2, 'number_of_days': 2})
 
         allocation_one.write({'number_of_days_display': 3, 'number_of_days': 3})
+
+    @users('admin')
+    @freeze_time('2024-03-25')
+    def test_allocation_dropdown_after_period(self):
+        """
+        Test when having two allocations of the same type with different
+        time range and submitting a request will the allocations be
+        shown correctly in the dropdown menu or not
+        :return:
+        """
+        leave = self.env.ref('hr_holidays.holiday_status_comp')
```

Comment:
It's a leave type, not a leave :innocent: 
```suggestion
        leave_type = self.env.ref('hr_holidays.holiday_status_comp')
```